pushsync/simulation_test.go

```diff
@@ -257,9 +258,12 @@ func upload(ctx context.Context, store Store, tags *chunk.Tags, tagname string,
 
 func download(ctx context.Context, store *storage.NetStore, addrs []storage.Address) error {
 	var g errgroup.Group
+	sem := make(chan struct{}, 32) // limit the number of concurrent gets
 	for _, addr := range addrs {
 		addr := addr
+		sem <- struct{}{}
 		g.Go(func() error {
+			defer func() { <-sem }()
 			_, err := store.Get(ctx, chunk.ModeGetRequest, storage.NewRequest(addr))
```

Comment:
Context is used incorrectly here - it has no timeout, which means that we have no global timeout for any retrieve requests.

I am not sure if `Netstore.Get` has the best interface, but it assumes that you set `some` timeout for the context.

Reply:
Not related to this PR, just making a mental node, maybe we want to change this in the future.